src/Shared/TypeLoader.cs

```diff
@@ -59,10 +59,10 @@ private static string[] findRuntimeAssembliesWithMicrosoftBuildFramework()
             string[] msbuildAssemblies = Directory.GetFiles(msbuildDirectory, "*.dll");
             string[] runtimeAssemblies = Directory.GetFiles(RuntimeEnvironment.GetRuntimeDirectory(), "*.dll");
 
-            List<string> msbuildAssembliesList = new(msbuildAssemblies);
-            msbuildAssembliesList.AddRange(runtimeAssemblies);
+            List<string> runtimeAssembliesList = new(runtimeAssemblies);
+            runtimeAssembliesList.AddRange(msbuildAssemblies);
 
-            return msbuildAssembliesList.ToArray();
+            return runtimeAssembliesList.ToArray();
```

Comment:
If we are going to return array anyways. Wouldn't it be better to create an array of length `msbuildAssemblies.Length + runtimeAssemblies.Length`? 
Please correct me if I am wrong, but wouldn't that be more efficient?

Reply:
You mean when we initially create the list?

That can be slightly more efficient. It shouldn't really matter, since this runs just once per process anyway, so the difference is negligible. I'd slightly prefer to keep it as-is because I think it's a bit easier to read.

Reply:
You are right, since it's going to run only once. I had no idea; I should be more active in the dotnet code-base repositories :).
Keeping it as is, is more readable indeed.